src/components/tx/ApprovalEditor/index.tsx

```diff
@@ -0,0 +1,179 @@
+import TokenIcon from '@/components/common/TokenIcon'
+import useBalances from '@/hooks/useBalances'
+import type { BaseTransaction } from '@safe-global/safe-apps-sdk'
+import { WarningOutlined } from '@mui/icons-material'
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
+
+import { Accordion, AccordionDetails, AccordionSummary, IconButton, Skeleton, Typography } from '@mui/material'
+import type { DecodedDataResponse, TokenInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import { ethers } from 'ethers'
+import { Interface, keccak256, parseUnits, toUtf8Bytes } from 'ethers/lib/utils'
+import { groupBy } from 'lodash'
+import css from './styles.module.css'
+import useAsync from '@/hooks/useAsync'
+import { getERC20TokenInfoOnChain, UNLIMITED_APPROVAL_AMOUNT } from '@/utils/tokens'
+import { ApprovalEditorForm } from './ApprovalEditorForm'
+import { useMemo } from 'react'
+
+const approvalSigHash = keccak256(toUtf8Bytes('approve(address,uint256)')).slice(0, 10)
+
+const erc20interface = new Interface(['function approve(address,uint256)'])
```

Comment:
Would we benefit from installing the ERC20 types?

Reply:
The method is ambiguous and causes trouble in ethers when I use the ERC20 interface.
The erc20 types from typechain are already available in the project.
Ethers v6 would solve this I think